tests/python/test_indices_assert.py

```diff
@@ -0,0 +1,15 @@
+import taichi as ti
+import pytest
+
+@ti.test(debug=True, gdb_trigger=False, exclude=[ti.metal])
```

Comment:
Let's just make sure it runs on CPU for now.

```suggestion
@ti.test(debug=True, gdb_trigger=False, arch=[ti.cpu])
```